### ISSUE TITLE
Provide suitable error messages if assertions are not met in `test_reader.py`

### ISSUE THREAD

**Issue by paddyroddy** [quantumjot/BayesianTracker#147]
See e.g. https://github.com/quantumjot/BayesianTracker/pull/115#discussion_r884937339

**Comment by dstansby** [comment]
I don't think we need a blanket policy of having messages with every assert statement in tests. For developers it should be pretty obvious from the grammar of the statement itself why it's failing.

**Comment by paddyroddy** [comment]
Personally, this has saved me a lot of time in the past when working with an unknown code base
